### ISSUE TITLE
Features don't work when selecting a Deleted object with checkbox

### ISSUE THREAD

**Issue by AlexZIX** [minio/console#2974]
I have bucket with versioning enabled. In this bucket I have the lifecycle policy which removes old versions after 7 days. But when I check the "Show deleted objects" checkbox, select deleted object and push the "Show  Object Versions" button it shows me the previous versions. Then I select v1 (NULL VERSION) and all buttons include "preview", "share", "download" becomes active and I can push it. "Preview" button opens the new window with "site can't be reached" error, "share" - new window with sharing options, "delete" doing nothing.

## Expected Behavior
All buttons with actions for objects versions which doesn't exists should be in inactive state. 

## Expected Behavior
Buttons active for v1 of removed object.

## Steps to Reproduce (for bugs)
1. Create versioned bucket.
2. Add lifetime policy which removes old versions after X days.
3. Add object.
4. Remove object.
5. Wait X days.
6. Select v1 of object.

## Your Environment
* MinIO version used (`minio --version`): 2023-07-07T07:13:57Z
* Server setup and configuration: 4 nodes with 4 drives on each
* Operating System and version (`uname -a`): Linux minio-cold-1 5.15.0-69-generic #76-Ubuntu SMP Fri Mar 17 17:19:29 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

**Comment by cesnietor** [comment]
We'll try to reproduce it, thanks for raising it.

**Comment by cesnietor** [comment]
I was able to reproduce this issue. 

The issue is: `Selecting deleted objects with checkbox shouldn't allow extra options other than Display Versions`

This is a bug on the way the object gets selected for versioned objects (and has nothing to do with lifecycle it seems)
The error occurs when selecting the object with `Checkbox` . If you click on the object itself it lets you download it just fine (by showing you first the versions). I was able to reproduce this both using and without lifecycle on a bucket.

Notes:
- When selecting multiple objects, including non deleted and deleted objects it shouldn't allow you to select deleted objects  cause they won'd be downloaded anyways.
- It should allow you to perform those actions on the version that is not marked as DELETED since that's why we preserve the old versions but this is a bug when selecting the object with a checkbox.

Test Steps:
- Create a versioned bucket
- Add an object
- Delete an object
- Show old versions
- Click on the checkbox of the deleted object
- [Error] Click Download or Preview and those fail



https://github.com/minio/console/assets/11819101/ae33d995-0239-4c6e-8ec8-615e1c34766b
